### PR TITLE
[Tests-Only] Added when step to create a user with any combination of attributes using the provisioning API

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
With this PR a when step is added to create a user using provisioning API with any combinations of attributes.

## Related Issue

- Part of https://github.com/owncloud/user_ldap/issues/561

## How Has This Been Tested?

- :robot: 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require an increase of minor version instead of the patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ public function theAdminCreatesAUserUsingTheWebUI(
 		$shouldExist = ($attemptTo === "creates");
 
 		$this->featureContext->addUserToCreatedUsersList(
-			$username, $password, "", $email, $shouldExist
+			$username, $password, $username, $email, $shouldExist
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php:151]: While creating users using webUI, username provided is set as user displayname, so we can set displayname as well in the createdUsersList

File: tests/acceptance/features/bootstrap/OCSContext.php [hunk 2]
```diff
@@ -130,6 +130,8 @@ public function userSendsHTTPMethodToOcsApiEndpointWithBody(
 		$bodyArray = [];
 		if ($body instanceof TableNode) {
 			$bodyArray = $body->getRowsHash();
+		} elseif ($body !== null && \is_array($body)) {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OCSContext.php:133]: PHPdoc for $body at line 115 could be:
```
 @param TableNode|array|null $body
```

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 3]
```diff
@@ -1060,6 +1060,38 @@ public function theAdministratorGetsAllEnabledAppsUsingTheProvisioningApi() {
 		$this->getEnabledApps();
 	}
 
+	/**
+	 * @When the administrator sends a user creation request with following attributes using the provisioning API:
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php:1064]:
```suggestion
	 * @When the administrator sends a user creation request with the following attributes using the provisioning API:
```

**Comment by kiranparajuli589** [comment]
@phil-davis comments has been addressed! Please review again!

**Review by phil-davis — APPROVED** [review]

**Pull request merged**